mldatafind/io.py

```diff
@@ -188,39 +196,78 @@ def read_timeseries(
     return data, times
 
 
+def _fetch_open_data(
+    ifos: List[str], start: float, end: float, verbose: bool, **kwargs
+) -> TimeSeriesDict:
+    ts_dict = TimeSeriesDict()
+    for ifo in ifos:
+        ts_dict[ifo] = TimeSeries.fetch_open_data(
+            ifo, start, end, verbose=verbose, **kwargs
+        )
+    return ts_dict
+
+
 def fetch_timeseries(
     channels: List[str],
-    t0: float,
-    tf: float,
-    nproc: int = 1,
+    start: float,
+    end: float,
     array_like: bool = False,
+    verbose: bool = False,
+    **kwargs,
 ) -> Union[TimeSeriesDict, Tuple[np.ndarray, np.ndarray]]:
     """
-    Fetch multiple channel timeseries from nds2 and store TimeSeriesDict,
+    Fetch multiple channel timeseries from nds2 and store in a TimeSeriesDict,
     or, if `array_lke` is True, a tuple of numpy arrays
     where the first element is an array of the channel data,
-    and the second element an array of corresponding times.
+    and the second element an array of corresponding times. If a channel name
+    is an interferometer name (e.g. `H1`), open data channels will be fetched.
+
     Thin wrapper around TimeSeriesDict.get
 
     Args:
         channels:
-            Channel names to fetch
-        t0:
+            List of channel names to fetch. If an interferometer
+            name is passed, (e.g. `H1`), open data channels will be fetched.
+        start:
             Start gpstime to read.
             If not passed will begin reading from earliest found time
-        tf:
+        end:
             Stop gpstime to read.
             If not passed will read until latest found time
-        nproc:
-            Number of concurrent processes to use with TimeSeriesDict.get
         array_like:
             Return in array like format. Otherwise, return gwpy.TimeSeriesDict
+        **kwargs:
+            key word arguments to pass to
+            TimeSeriesDict.get or TimeSeries.fetch_open_data
 
     Returns gwpy.TimeSeriesDict or Tuple of np.ndarrays
     """
+
+    open_data_channels = list(
+        filter(lambda x: x in OPEN_DATA_CHANNELS, channels)
+    )
+    channels = list(filter(lambda x: x not in OPEN_DATA_CHANNELS, channels))
+
+    # fetch data from nds2
     ts_dict = TimeSeriesDict.get(
-        channels, start=t0, end=tf, nproc=nproc, verbose=True
+        channels,
```

Comment:
What will happen here if you've only requested open data channels, does `TimeSeriesDict.get` just return an empty dict if `channels=[]`? For the benefit of clarity, would it make more sense to initialize `ts_dict = {}` and do a similar `if channels:` clause here?

For that matter, does it make sense to just use `get` for _all_ the channels if there are _any_ non-open channels so that it can all be handled in one call in parallel, and only use `_fetch_open_data` if all the channels are open?

Reply:
1. Good catch. I imagine `TimeSeriesDict.get` will throw an exception but not completely sure. Will address this / write a unit test.
2. Not sure I fully understand. My assumption (that might be wrong) was that the only way to get the open data was via `fetch_open_data`. I expect that calling `.get` with open data channels will throw an exception.

Reply:
Turns out passing empty channel list will return an empty TimeSeriesDict. Still added the `if channels:` for clarity